Lesson 2

NadezhdaPenkrat opened the pull request:
Hello! Check, please

NadezhdaPenkrat committed:
upd

NadezhdaPenkrat committed:
upd

NadezhdaPenkrat committed:
additionaltasks

NadezhdaPenkrat committed:
methodicaltasks

TheFallen058 commented on lesson_2/lesson_2.iml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Please add iml files to git ignore
and remove them from repositiry

TheFallen058 replied:
Please add iml files to git ignore
and remove them from repository

TheFallen058 commented on lesson_2/src/com/freeit/additionaltasks/Task1VarNum.java:
```diff
@@ -0,0 +1,37 @@
+ Достаточно будет определить, является ли число однозначным,
+ "двухзначным или трехзначным и более.
+ */
+
```
Please use 'Reformat code' option

TheFallen058 commented on lesson_2/src/com/freeit/additionaltasks/Task1VarNum.java:
```diff
@@ -0,0 +1,37 @@
+        int number = random.nextInt(1000);
+        System.out.println ( " Number " +number );
+
+        if (number>-10 && number <0) {
```
it's better to check if number is positive or not
then check amount of digits
and print results

TheFallen058 commented on lesson_2/src/com/freeit/additionaltasks/Task3NumCalculation.java:
```diff
@@ -0,0 +1,26 @@
+        int number = random.nextInt(1000);
+        System.out.println(" Дано целое число: " +number);
+        if (number > 0) {
+            if (number > 0) {
```
Why do you check it twice?

TheFallen058 replied:
Why do you check it twice?

TheFallen058 commented on lesson_2/src/com/freeit/additionaltasks/Task3NumCalculation.java:
```diff
@@ -0,0 +1,26 @@
+        System.out.println(" Дано целое число: " +number);
+        if (number > 0) {
+            if (number > 0) {
+                System.out.println("Если оно является положительным, то прибавить к нему 1:  " + (number = number + 1));
```
you don't need "number =" in this case as you don't use it's result anymore

TheFallen058 commented on lesson_2/src/com/freeit/additionaltasks/Task3NumCalculation.java:
```diff
@@ -0,0 +1,26 @@
+            } else {
+                System.out.println("Если нулевым, то заменить его на 10:  " + ( number = 10));
+            }
+        }
```
should be something like this:
if (number > 0) {
                System.out.println("Если оно является положительным, то прибавить к нему 1:  ");
                number = number + 1;
            } else if (number < 0) {
                System.out.println("Если отрицательным, то вычесть из него 2:  ");
                number = number - 2;
            } else {
                System.out.println("Если нулевым, то заменить его на 10:  ");
                number = 10;
            }
            System.out.println(number);

TheFallen058 commented on lesson_2/src/com/freeit/additionaltasks/Task5FindPositiveNegative.java:
```diff
@@ -0,0 +1,35 @@
+public class Task5FindPositiveNegative {
+    public static void main(String[] args) {
+        Random random = new Random();
+        int firstNumber = random.nextInt(1000);
```
random.nextInt returns only positive numbers)

TheFallen058 commented on lesson_2/src/com/freeit/additionaltasks/Task13OutputtingNum.java:
```diff
@@ -0,0 +1,13 @@
+// 13)Напишите программу вывода всех четных чисел от 2 до 100 включительно
+public class Task13OutputtingNum {
+    public static void main(String[] args) {
+        int[] array = new int[21];
```
Why do you need this array? it's empty and you do nothing with its values

TheFallen058 commented on lesson_2/src/com/freeit/methodicaltasks/Task6NumDigEven.java:
```diff
@@ -0,0 +1,41 @@
+ */
+public class Task6NumDigEven {
+    public static void main(String[] args) {
+        Random random = new Random();
```
you don't use it

TheFallen058 commented on lesson_2/src/com/freeit/methodicaltasks/Task9FactorialComposition.java:
```diff
@@ -0,0 +1,27 @@
+        int number = random.nextInt(15);
+        int factor = 1;
+        System.out.println(number * factor);
+        int result = 1;
```
you do nothing with result

TheFallen058 commented on lesson_2/src/com/freeit/methodicaltasks/Task11NumPrime.java:
```diff
@@ -0,0 +1,24 @@
+                break;
+            }
+        }
+        if (isComposite) {
```
Without printing the number not clear if program works correct

TheFallen058 commented on lesson_2/src/com/freeit/methodicaltasks/Task14MaxMin.java:
```diff
@@ -0,0 +1,32 @@
+        int minimum = 0;
+
+        int[] array = new int[10];
+        System.out.println("array"+ array);
```
You can't print array values such way

TheFallen058 commented on lesson_2/src/com/freeit/methodicaltasks/Task14MaxMin.java:
```diff
@@ -0,0 +1,32 @@
+                maximum = array[i];
+            }
+        }
+        System.out.println("Max"+ maximum );
```
No index of min and max numbers in result

TheFallen058 commented on lesson_2/src/com/freeit/methodicaltasks/Task16ArrayNumBetween.java:
```diff
@@ -0,0 +1,104 @@
+
+    public static void main(String[] args) {
+        Random random = new Random();
+        int n = random.nextInt();
```
You don't use it

TheFallen058 commented on lesson_2/src/com/freeit/methodicaltasks/Task17ArrayButterfly.java:
```diff
@@ -0,0 +1,40 @@
+public class Task17ArrayButterfly {
+
+    public static void main(String[] args) {
+            int arr[][]=new int[5][5];
```
please use java style array initialisation
int[][] arr

TheFallen058 reviewed (approved):
Mostly well done
good variable naming

NadezhdaPenkrat committed:
use java style array initialisation
int[][] arr

NadezhdaPenkrat committed:
remote don't needed array

NadezhdaPenkrat committed:
changed:
checked number is positive or not
then checked amount of digits
and print results

NadezhdaPenkrat committed:
remote double.fixed

NadezhdaPenkrat committed:
added negative numbers option

NadezhdaPenkrat committed:
removed unused syntax

NadezhdaPenkrat committed:
upd

NadezhdaPenkrat committed:
Merge branch 'main' into lesson_2
